test/DynamoCoreTests/CodeBlockNodeTests.cs

```diff
@@ -339,6 +339,43 @@ public void TestVarRedefinitionInFunctionDef()
             AssertPreviewValue(guid, 3);
         }
 
+        [Test]
+        [Category("UnitTests")]
+        public void TestOutputPortUpdateWithFunctionDef()
+        {
+            var cbn = CreateCodeBlockNode();
+            UpdateCodeBlockNodeContent(cbn, 
+                "def foo ( a : double[], b :double[] )" +
+                "{" +
+                    "return = Count(a) + Count(b);" +
+                "}" + 
+                "a = [ 1, 2 ];" +
+                "b = [3, 4, 5];" +
+                "test = foo (a, b);"
+                );
+
+            Assert.IsNotNull(cbn);
+
+            Assert.IsTrue(cbn.CodeStatements.Any());
+
+            Assert.AreEqual(0, cbn.InPorts.Count);
+            Assert.AreEqual(3, cbn.OutPorts.Count);
+            Assert.AreEqual(0, cbn.AllConnectors.Count());
+            AssertPreviewValue(cbn.GUID.ToString(), 5);
+
+           UpdateCodeBlockNodeContent(cbn, "def foo ( a : double[], b :double[] )" +
+                                           "{" +
+                                           "return = Count(a) + Count(b);" +
+                                           "}");
+           Assert.IsNotNull(cbn);
+
+           Assert.AreEqual(1, cbn.CodeStatements.Count());
+
+           Assert.AreEqual(0, cbn.InPorts.Count);
+           Assert.AreEqual(0, cbn.OutPorts.Count);
+           Assert.AreEqual(0, cbn.AllConnectors.Count());
+        }
+
```

Comment:
Can we add a test for a case in which we have the same initial state but after the edit we also have the assignment `a = [ 1, 2 ];`?

Reply:
@mmisol done.

Reply:
Nice. Thanks!